File: Lista4/exec3/forma.py
class forma:

    def __init__(self, cheio: bool, cor: str):
        self.cheio = cheio
        self.lados = [0, 0]
        if(self.cheio):
            self.cheio = cor
        
    
    def forma(self, lado1: float, lado2: float, cheio: bool, cor: str):
        self.cheio = cheio
        self.lados[0] = lado1
        self.lados[1] = lado2
        if(self.cheio):
            self.cor = cor

    def perimetro(self):
        return 2*(lados[0] + laddos[1])

    def area(self):
        return lados[0]*lados[1]

    # Getters
    def getLados(self):
        return self.lados
    def getCheio(self):
        return self.cheio
    def getCor(self):
        return self.cor

File: Lista3/Lista3_11800910/exec3/Forma.java
/**
 * Classe que cria a forma geométrica
 */
public class Forma {

    private double[] lados = new double[2];
    private boolean cheio;
    private String cor;

    /**
     * Construtor para adicionar apenas cor
     * @param cheio
     * @param cor
     */
    public Forma(boolean cheio, String cor){
        this.cheio = cheio;
        if(this.cheio){ 
            this.cor = cor; } 
    }

    /**
     * Construtor para adicionar cor e lados
     * @param lado1
     * @param lado2
     * @param cheio
     * @param cor
     */
    public Forma(double lado1, double lado2, boolean cheio, String cor){

        this.cheio = cheio;
        this.lados[0] = lado1;
        this.lados[1] = lado2;
        if(this.cheio){ 
            this.cor = cor; 
        } 
    }

    /**
     * Cálculo de área
     * @return
     */
    public double perimetro(){
        return 2 * (lados[0] + lados[1]);
    }

    /**
     * Cálculo de perimetro
     * @return
     */
    public double area(){

        return lados[0] * lados[1];
    }

    //Getters
    public double[] getLados(){ 
        return this.lados; 
    }
    public boolean getCheio(){ 
        return this.cheio; 
    }
    public String getCor(){ 
        return this.cor; 
    }

}